test/distributed/test_data_parallel.py

```diff
@@ -630,6 +630,24 @@ def test_replicate_buffers(self):
                 self.assertEqual(replica.bn.running_var.get_device(), i, 'buffer on wrong device')
                 self.assertEqual(replica.bn.num_batches_tracked.get_device(), i, 'buffer on wrong device')
 
+    @unittest.skipIf(not TEST_MULTIGPU, "multi-GPU not supported")
+    def test_zero_grad(self):
+        # zero_grad should warn about using gradients inside forward
+
+        class Net(torch.nn.Module):
+            def __init__(self, testcase):
+                super(Net, self).__init__()
+                self._testcase = testcase
+
+            def forward(self, x):
+                self._testcase.assertWarnsRegex(
+                    lambda: self.zero_grad(),
+                    "The parameters in data parallel modules are copied from the original module.")
```

Comment:
The test fails because I changed the error message in the last commit. Fixed in #33064